src/core/MIGRATION.md

```diff
@@ -848,3 +849,43 @@ class MyPlugin {
     this.config$ = initializerContext.config.createIfExists();
   }
 ```
+
+### Mock core services in tests
+Core services already provide mocks to simplify testing and make sure plugins always rely on valid public contracts. 
```

Comment:
Can you add some blank lines in between these sections to make them a little more readable?

Reply:
you mean code or text?